aerobites/cypress/e2e/main_page.cy.js

```diff
@@ -13,4 +13,77 @@ describe('Display headere on page load and drop down of airports', () => {
       .get('header').should('contain', 'AeroBites')
   })
 
-})
\ No newline at end of file
+  it('should display the "Show Favorites" button', () => {
+    cy.get('#show-favorites button').should('be.visible');
+  });
+
+//testing dropdown 
+it('should display a dropdown with airports', () => {
+  cy.get('.airports-select').should('be.visible');
+  cy.get('.airports-select').children('option').should('have.length', 7); // my mock data and the select airport dummy option
+});
+
+
+})
+
+
+
+//Testing the terminals and business 
+
+describe('Airport Details', () => {
+  beforeEach(() => {
+      // Intercept the initial airports API call
+      cy.intercept('GET', 'http://localhost:8080', { 
+        fixture: 'airportdata.json' }).as('getAirports');
+
+      // Intercept calls for terminals and businesses
+      cy.intercept('GET', 'http://localhost:8080/terminals', { 
+        fixture: 'terminalsdata.json' }).as('getTerminals');
+      cy.intercept('GET', 'http://localhost:8080/businesses', { 
+        fixture: 'businessesdata.json' }).as('getBusinesses');
+
+      // Visit the main page and wait for the airports to load
+      cy.visit('http://localhost:3001');
+      cy.wait('@getAirports');
+  });
+
+  it('should fetch and display terminals and businesses for a selected airport', () => {
+      // Simulate selecting an airport
+      cy.get('.airports-select').select('Hartsfield-Jackson Atlanta International Airport');
+
+      // Wait for the API calls to complete
+      cy.wait('@getTerminals');
+      cy.wait('@getBusinesses');
+
+      // Check if terminals and businesses are rendered correctly
+    
+      cy.get('.terminals-container').should('exist');
+      cy.get('.businesses-container').should('exist');
+
+      // Check if specific terminals and businesses are rendered correctly
+      cy.contains('h3', 'Domestic Terminal').should('exist');
```

Comment:
I'm assuming these are lists of terminals and restaurants - test lists with first() and last(), as well as every data point that is displayed